System/ContentFile.cpp

```diff
@@ -180,107 +126,117 @@ namespace RTE {
+			RTEAbort("Failed to find audio file with following path and name:\n\n" + m_DataPathAndReaderPosition);
+		} else {
+			std::string errorMessage;
+
+			unsigned int fileSize = static_cast<unsigned int>(std::experimental::filesystem::file_size(m_DataPath));
+			PACKFILE *fileToLoad = pack_fopen(m_DataPath.c_str(), F_READ);
 
-				// Deallocate the intermediary data and close the file stream
-				delete[] rawData;
-				pack_fclose(pFile);
-			} else if (separatorPos != m_DataPath.length() - 1) {
-				RTEAbort("Loading sounds from allegro datafiles isn't supported yet!");
-				/*
-				// Split the datapath into the path and the object name and load the datafile from them
-				m_DataFile = load_datafile_object(m_DataPath.substr(0, separatorPos).c_str(), m_DataPath.substr(separatorPos + 1).c_str());
-				RTEAssert(m_DataFile && m_DataFile->dat && m_DataFile->type == DAT_BITMAP, "Failed to load datafile object with following path and name:\n\n" + m_DataPath);
-
-				returnSample = (FMOD::Sound *)m_DataFile->dat;
-				*/
+			if (!fileToLoad || fileSize <= 0) {
+				RTEAssert(!abortGameForInvalidSound, "Failed to load audio file with following path and name:\n\n" + m_DataPathAndReaderPosition);
+				g_ConsoleMan.AddLoadWarningLogEntry(m_DataPath, GetCurrentlyReadFileAndLine());
+				return returnSample;
 			}
-			if (!returnSample) {
-				errorMessage = "Failed to load sound file with following path and name:";
-				if (abortGameForInvalidSound) { RTEAbort(errorMessage + "\n\n" + m_DataPath); }
-				g_ConsoleMan.PrintString("Error: " + errorMessage + m_DataPath);
+			char *rawData = new char[fileSize];
+			int bytesRead = pack_fread(rawData, fileSize, fileToLoad);
+			RTEAssert(bytesRead == fileSize, "Read audio file data does not match the reported size of file with following path and name:\n\n" + m_DataPathAndReaderPosition + "\nThe file may be corrupt.");
+			pack_fclose(fileToLoad);
+
+			// Setup fmod info, and make sure to use mode OPENMEMORY since we're doing the loading with ContentFile instead of fmod, and we're deleting the raw data after loading it
+			FMOD_CREATESOUNDEXINFO soundInfo = {};
+			soundInfo.cbsize = sizeof(FMOD_CREATESOUNDEXINFO);
+			soundInfo.length = fileSize;
+			//TODO Consider doing FMOD_CREATESAMPLE for dumping audio files into memory and FMOD_NONBLOCKING to async create sounds
+			FMOD_RESULT result = g_AudioMan.GetAudioSystem()->createSound(rawData, FMOD_OPENMEMORY | FMOD_3D, &soundInfo, &returnSample);
+			delete[] rawData;
+
+			if (result != FMOD_OK) {
+				errorMessage = "Unable to create sound because of FMOD error: " + std::string(FMOD_ErrorString(result)) + "\nThe path and name were: ";
+				RTEAssert(!abortGameForInvalidSound, errorMessage + "\n\n" + m_DataPathAndReaderPosition);
+				g_ConsoleMan.PrintString("ERROR: " + errorMessage + m_DataPath);
+				g_ConsoleMan.AddLoadWarningLogEntry(m_DataPath, GetCurrentlyReadFileAndLine());
```

Comment:
So you know, the only time sound loading will be done with `abortGameForInvalidSound = false` right now is when someone does it through lua. Loading from ini will always have `abortGameForInvalidSound = true` so I'm not sure it makes sense to add a load warning log entry. At the same time, no harm in it, and this may change in future.